src/libraries/System.Private.CoreLib/src/System/Action.cs

```diff
@@ -23,7 +23,10 @@ namespace System
     public delegate void Action<in T1, in T2, in T3, in T4, in T5, in T6, in T7, in T8, in T9, in T10, in T11, in T12, in T13, in T14, in T15>(T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6, T7 arg7, T8 arg8, T9 arg9, T10 arg10, T11 arg11, T12 arg12, T13 arg13, T14 arg14, T15 arg15);
     public delegate void Action<in T1, in T2, in T3, in T4, in T5, in T6, in T7, in T8, in T9, in T10, in T11, in T12, in T13, in T14, in T15, in T16>(T1 arg1, T2 arg2, T3 arg3, T4 arg4, T5 arg5, T6 arg6, T7 arg7, T8 arg8, T9 arg9, T10 arg10, T11 arg11, T12 arg12, T13 arg13, T14 arg14, T15 arg15, T16 arg16);
 
-    public delegate int Comparison<in T>([AllowNull] T x, [AllowNull] T y);
+    // This should probably technically be T? rather than T to match `IComparer<T>`.  However, the use cases are generally different,
+    // with Comparison<T> typically being used via a lambda, with the T inferred from the type of the collection being sorted,
+    // and forcing nullable warnings onto all such usage leads to many spurious warnings.
+    public delegate int Comparison<in T>(T x, T y);
```

Comment:
Note this change.  It turns out Roslyn hasn't been validating the [AllowNull] here (https://github.com/dotnet/roslyn/issues/46459).  When I switched from `[AllowNull]T` to `T?`, we started getting warnings elsewhere because of code like:
```C#
List<string> list = ...;
list.Sort((x, y) => x.CompareTo(y));
```
so while technically to match `IComparer<T>` this should use `T?`, I believe it's better to use `T`.  If others disagree, obviously it can be changed.

Reply:
Interesting, just tried it with `string?` and the compiler infers the nullability correctly. So seems like a reasonable change to me. 

https://sharplab.io/#v2:CYLg1APgAgTAjAWAFDIMQDsCuAbbBDAI2wFMACY9Qk5KAZlNlIGFSBvZUzh+gS3QBdSAWQA8AFQB8ACiYB7ALYAHPACceAZ1npxE0gGMFytZvQAaMaQAep0hYCeASjYcurqAHZ9h1Rq1TrpI4A3C6cAL6hpJF0pMDEJADmePxkfIJySj4mIny20hYB9g4hSK7R9FAALKQAYrKyUg6R7KWuXGlWpAC8wiJQcAAMAPzS/jaO3bqWQwB0GUbEYg0TQ0OkAzYARAObpKabcJvFkRFIYUA===